src/main/java/fm/last/moji/impl/MojiFileAttributesImpl.java

```diff
@@ -20,6 +21,12 @@
     deviceCount = Integer.parseInt(valueMap.get("devcount"));
     storageClass = valueMap.get("class");
     key = valueMap.get("key");
+
+    if (valueMap.containsKey("checksum")) {
+      checksum = valueMap.get("checksum");
+    } else {
```

Comment:
No need for the else statement, checksum will be initialised to null by default. Having said that you don't actually even need the if statement at all, the contract for java.util.Map get() says "Returns the value to which the specified key is mapped, or null if this map contains no mapping for the key." so this code can go back to how it was before.